addons/binding/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/connector/http/DenonMarantzHttpConnector.java

```diff
@@ -0,0 +1,410 @@
+import org.openhab.binding.denonmarantz.internal.xml.entities.commands.AppCommandRequest;
+import org.openhab.binding.denonmarantz.internal.xml.entities.commands.AppCommandResponse;
+import org.openhab.binding.denonmarantz.internal.xml.entities.commands.CommandRx;
+import org.openhab.binding.denonmarantz.internal.xml.entities.commands.CommandTx;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class makes the connection to the receiver and manages it.
+ * It is also responsible for sending commands to the receiver.
+ * *
+ *
+ * @author Jeroen Idserda - Initial Contribution (1.x Binding)
+ * @author Jan-Willem Veldhuis - Refactored for 2.x
+ */
+public class DenonMarantzHttpConnector extends DenonMarantzConnector {
+
+    private Logger logger = LoggerFactory.getLogger(DenonMarantzHttpConnector.class);
+
+    private static final int REQUEST_TIMEOUT_MS = 5000; // 5 seconds
+
+    // Main URL for the receiver
+    private static final String URL_MAIN = "formMainZone_MainZoneXml.xml";
+
+    // Main Zone Status URL
+    private static final String URL_ZONE_MAIN = "formMainZone_MainZoneXmlStatus.xml";
+
+    // Secondary zone lite status URL (contains less info)
+    private static final String URL_ZONE_SECONDARY_LITE = "formZone%d_Zone%dXmlStatusLite.xml";
+
+    // Device info URL
+    private static final String URL_DEVICE_INFO = "Deviceinfo.xml";
+
+    // URL to send app commands to
+    private static final String URL_APP_COMMAND = "AppCommand.xml";
+
+    private static final String CONTENT_TYPE_XML = "application/xml";
+
+    private final String cmdUrl;
+
+    private final String statusUrl;
+
+    private final HttpClient httpClient;
+
+    private ScheduledFuture<?> pollingJob;
+
+    public DenonMarantzHttpConnector(DenonMarantzConfiguration config, DenonMarantzState state,
+            ScheduledExecutorService scheduler) {
+        this.config = config;
+        this.scheduler = scheduler;
+        this.state = state;
+        this.cmdUrl = String.format("http://%s:%d/goform/formiPhoneAppDirect.xml?", config.getHost(),
+                config.getHttpPort());
+        this.statusUrl = String.format("http://%s:%d/goform/", config.getHost(), config.getHttpPort());
+        this.httpClient = new HttpClient();
+    }
+
+    public DenonMarantzState getState() {
+        return state;
+    }
+
+    /**
+     * Set up the connection to the receiver by starting to poll the HTTP API.
+     */
+    @Override
+    public void connect() {
+        refreshState();
+        startPolling();
+    }
+
+    private void startPolling() {
+        if (!isPolling()) {
+            logger.debug("HTTP polling started.");
+            pollingJob = scheduler.scheduleWithFixedDelay(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        refreshHttpProperties();
+                    } catch (RuntimeException e) {
```

Comment:
I just saw your discussion and your comment

> So I think I still need to catch RuntimeException to prevent the scheduler from stopping. Will perform some testing around this.

I hope your testing showed that it is ok to stop the scheduled job here - if the problem is due to a configuration error, the job should be rather re-scheduled once the configuration is fixed.

Reply:
Well the main reason was the exception was nowhere to be found and was very very difficult to diagnose and debug due to this. What if 20 channels are fine and only 1 channel is mistakenly misconfigured (e.g. Dimmer type used for Item instead of Number type), not sure whether we should stop everything? At least the error should be catched and logged..

How to determine whether the configuration has been fixed so polling can resume?

Reply:
Could I use the `channelLinked` as trigger to resume polling? E.g. when one has updated an Item definition? Reconfiguration of the Thing itself currently restarts everything already.
The only known cause until now for the RunTimeException was a malformed manual Item definition in a `.items` file.

Reply:
BTW is there a way to warn the user he/she has mapped a `Dimmer` item to a `Number` channel? It can work for a while, but at some point you can get nasty errors. The error thrown is not very helpful in telling the user it was their fault and they should look into the item definition.

Reply:
@kaikreuzer This remains an open item. Any suggestion on how to trigger a resume of the polling after it failed?

Reply:
Sorry, @jwveldhuis, I missed to follow up on your questions.

> BTW is there a way to warn the user he/she has mapped a Dimmer item to a Number channel? 

No, I actually consider it a bug in the framework that it is at all possible (and that handlers are affected by it), see https://github.com/eclipse/smarthome/issues/2666.

To come to a solution here: Let's blame the framework for throwing RuntimeExceptions here and let's keep the code as it is for now.